Tests/kaas/k8s-version-recency/k8s-version-recency-check.py

```diff
@@ -0,0 +1,369 @@
+        with open(config_path, "r") as fd:
+            config_file = yaml.safe_load(fd)
+    except OSError:
+        pass
+
+    if config_file:
+        # Setup logging if the config file with the relevant information could be loaded before
+        setup_logging(config_file['log'])
+    else:
+        # Otherwise, we initialize basic logging
+        logging.basicConfig(format='%(levelname)s: %(message)s', level=logging.INFO)
+
+    if not config.kubeconfig:
+        print("A kubeconfig needs to be set in order to test a k8s cluster version.")
+        raise ConfigException
+
+    return config
+
+
+class K8sVersionInfo:
+    """Class that contains a k8s version info.
+
+    Attributes:
+        major (int): Major version of the k8s version
+        minor (int): Minor version of the k8s version
+        patch (int): Patch version of the k8s version
+        date (datetime): release date of the k8s version
+    """
+    def __init__(self, major=0, minor=0, patch=0):
+        self.major = major
+        self.minor = minor
+        self.patch = patch
+
+        self.date = None
+
+    def __eq__(self, other):
+        if not isinstance(other, K8sVersionInfo):
+            return False
+        return self.major == other.major and self.minor == other.minor and self.patch == other.patch
+
+    @classmethod
+    def extract_version(cls, string, separator="."):
+        components = string.strip().split(separator)
+        return cls(int(components[0]), int(components[1]), int(components[2]))
+
+    def check_for_version(self, major=None, minor=None, patch=None):
+        return (major is None or self.major == major) and \
+               (minor is None or self.minor == minor) and \
+               (patch is None or self.patch == patch)
+
+    def __str__(self):
+        return f"{self.major}.{self.minor}.{self.patch}"
+
+
+def diff_months(date1, date2):
+    return abs((date1.year - date2.year) * 12 + date1.month - date2.month)
+
+
+def diff_weeks(date1, date2):
+    day1 = (date1 - datetime.timedelta(days=date1.weekday()))
+    day2 = (date2 - datetime.timedelta(days=date2.weekday()))
+    diff = day2 - day1
+    return abs((diff.days / 7) + math.ceil(diff.seconds / 86400))
+
+
+def diff_days(date1, date2):
+    delta = date1 - date2
+    return abs(delta.days)
+
+
+async def request_cve_data(session: aiohttp.ClientSession, cveid: str) -> dict:
+    """Request for a single CVE data item.
+    """
+    resp = await session.request('GET', f"https://cveawg.mitre.org/api/cve/{cveid}",
+                                 headers={"Accept": "application/json"})
+    return await resp.json()
+
+
+async def collect_cve_versions():
+    """Get all relevant CVE versions, that are relevant for the test according to the severity
+    dictated by the Standard.
+    """
+
+    # CVE fix versions
+    cfvs = list()
+
+    # Request latest version
+    cve_list = requests.get("https://kubernetes.io/docs/reference/issues-security/official-cve-feed/index.json",
```

Comment:
I wouldn't mix `requests` and `aiohttp` in the same program. Also, I think `collect_cve_versions` should take a ClientSession as argument just like `request_cve_data` does.

Reply:
Added the ClientSession as an argument. Also tried to follow the `requests`/`aiohttp` advise.